x-pack/plugins/ml/public/application/components/controls/checkbox_showcharts/checkbox_showcharts.tsx

```diff
@@ -4,41 +4,43 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-/*
- * React component for a checkbox element to toggle charts display.
- */
-import React, { FC } from 'react';
-
-import { EuiCheckbox } from '@elastic/eui';
-// @ts-ignore
-import makeId from '@elastic/eui/lib/components/form/form_row/make_id';
-
+import React, { FC, useCallback, useMemo } from 'react';
+import { EuiCheckbox, htmlIdGenerator } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-
-import { useUrlState } from '../../../util/url_state';
+import { useExplorerUrlState } from '../../../explorer/hooks/use_explorer_url_state';
 
 const SHOW_CHARTS_DEFAULT = true;
-const SHOW_CHARTS_APP_STATE_NAME = 'mlShowCharts';
 
-export const useShowCharts = () => {
-  const [appState, setAppState] = useUrlState('_a');
+export const useShowCharts = (): [boolean, (v: boolean) => void] => {
+  const [explorerUrlState, setExplorerUrlState] = useExplorerUrlState();
+
+  const showCharts = explorerUrlState?.mlShowCharts ?? SHOW_CHARTS_DEFAULT;
 
-  return [
-    appState?.mlShowCharts !== undefined ? appState?.mlShowCharts : SHOW_CHARTS_DEFAULT,
-    (d: boolean) => setAppState(SHOW_CHARTS_APP_STATE_NAME, d),
-  ];
+  const setShowCarts = useCallback(
```

Comment:
Typo - should be `setShowCharts` I guess!?

Reply:
Fixed in https://github.com/elastic/kibana/pull/83792/commits/77bc9c074e76f7c341f843de4c75f385a19f011e